cep-9999.md

```diff
@@ -0,0 +1,80 @@
+# CEP ???? - Names in conda packages and channels
+
+<table>
+<tr><td> Title </td><td> CEP ???? - Names in conda packages and channels </td>
+<tr><td> Status </td><td> Draft </td></tr>
+<tr><td> Author(s) </td><td> Jaime Rodríguez-Guerra &lt;jaime.rogue@gmail.com&gt; <br /> Matthew R. Becker &lt;becker.mr@gmail.com&gt;</td></tr>
+<tr><td> Created </td><td> Mar 11, 2025</td></tr>
+<tr><td> Updated </td><td> Mar 11, 2025</td></tr>
+<tr><td> Discussion </td><td> N/A </td></tr>
+<tr><td> Implementation </td><td> N/A </td></tr>
+</table>
+
+## Abstract
+
+This CEP aims to standardize names and other strings used throughout the conda ecosystem.
+
+## Specification
+
+### Artifacts
+
+The conda ecosystem distinguishes between two types of packages:
+
+- Regular package names: represented by a concrete, downloadable, extractable conda artifact.
+- Virtual package names: not backed by any physical artifact. They only exist on the client side.
+
+### Package names
+
+A regular package name MUST only consist of lowercase ASCII letters, numbers, hyphens, periods and underscores. It MUST start with a letter, a number, or a single underscore. It MUST NOT start with two consecutive underscores.
+
+Virtual package names MUST only consist of lowercase ASCII letters, numbers, hyphens, periods and underscores. They MUST start with two underscores.
+
+This means that regular package names are limited to those which match the following case-insensitive regex: `^(([a-z0-9])|([a-z0-9_](?!_)))[._-]?([a-z0-9]+(\.|-|_|$))*$`.
+
+Virtual package names MUST follow this other regex: `^__([a-z0-9]+(\.|-|_|$))+$`.
+
+### Version strings
+
+Version strings MUST only consist of digits, periods, lowercase ASCII letters, underscores, hyphens, plus signs and exclamation marks. Hyphens and underscores MAY be used interchangeably, but they MUST NOT be mixed in the same version string.
+
+Version strings MAY specify an epoch field, which MUST be an integer separated by a single exclamation mark from the rest of the version information. If not specified, the epoch is considered to be `0`. The exclamation mark MUST NOT be used anywhere else.
+
+The version string SHOULD start with a digit.
+
+The plus sign SHOULD separate the regular version information from the local version information.
+The plus sign SHOULD NOT be used for any other purposes.
+
+Multiple separator characters (periods, underscores, plus signs) SHOULD NOT be used contiguously.
+
+The following regex does not cover all rules, but all valid version strings MUST match it: `^([0-9]+!)?[0-9][0-9a-z.(_|-)+]*$`.
```

Comment:
This is probably incomplete, but it's a beginning. This does NOT standardize version _specifiers_ (e.g. `>=2.0` or `1.7.*`), just version _strings_ (`2.0.0` and `1!1.3.10post1+h12345`).